src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs

```diff
@@ -466,7 +524,44 @@ private static Task GetHostEntryOrAddressesCoreAsync(string hostName, bool justR
             if (NameResolutionPal.SupportsGetAddrInfoAsync && ipAddress is null)
             {
                 ValidateHostName(hostName);
-                return NameResolutionPal.GetAddrInfoAsync(hostName, justAddresses);
+
+                if (NameResolutionTelemetry.IsEnabled)
+                {
+                    ValueStopwatch stopwatch = NameResolutionTelemetry.Log.ResolutionStart(hostName);
+
+                    Task coreTask;
+                    try
+                    {
+                        coreTask = NameResolutionPal.GetAddrInfoAsync(hostName, justAddresses);
+                    }
+                    catch when (LogFailure(hostName, stopwatch))
+                    {
+                        Debug.Fail("LogFailure should return false");
+                        throw;
+                    }
+
+                    coreTask.ContinueWith(
```

Comment:
Why not use async await?

Reply:
Marking as resolved. See comment below https://github.com/dotnet/runtime/pull/38409#discussion_r447076076.